### DISCUSSION TITLE
Refactor Maven and the mostly used plugins to be reproducible builds

### DISCUSSION THREAD

**Discussion by hendrikebbers** [support-and-care/maven-support-and-care#7] [Ideas for Funding]
We need to check if Maven and the mostly used plugins are support reproducible builds. If not we should refactor the project (and maven its dependencies if needed) to support reproducible builds. Best would be to have an automatic check as one of the release steps for Maven.

**Comment by khmarbaise** [comment]
Some references: 
* https://reproducible-builds.org/
* https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=74682318

**Comment by aalmiray** [comment]
There is also https://github.com/jvm-repo-rebuild/reproducible-central

**Reply by hendrikebbers** [reply]
Nice. Missed that one

**Comment by hboutemy** [comment]
> Best would be to have an automatic check as one of the release steps for Maven.

see command at the end of "Call the vote" section https://maven.apache.org/developers/release/maven-project-release-procedure.html#call-the-vote

notice that Reproducible Central detects staged Apache releases (of already tested projects)  and proposes a staging rebuild https://github.com/jvm-repo-rebuild/reproducible-central/blob/master/doc/add.md#apache-release-candidate-in-staging

**Reply by khmarbaise** [reply]
That is related to #27 to handle that in an automated way...

**Comment by hboutemy** [comment]
FYI, 4 years ago, I did such checks on SNAPSHOTs of "everything Maven": https://github.com/jvm-repo-rebuild/reproducible-maven-HEAD
I just archived it as every Maven-related release is checked for a few years now: this repository is just a view on how I did the global preparation

**Comment by ascheman** [comment]
This discussion has been converted to Epic #68 for tracking.

Closing this discussion.
